ext/random/random.c

```diff
@@ -536,10 +533,12 @@ PHPAPI int php_random_bytes(void *bytes, size_t size, bool should_throw)
 		              amount_to_read
 		*/
 		size_t amount_to_read = size - read_bytes;
+
+		errno = 0;
 #  if defined(__linux__)
-		n = syscall(SYS_getrandom, bytes + read_bytes, amount_to_read, 0);
+		ssize_t n = syscall(SYS_getrandom, bytes + read_bytes, amount_to_read, 0);
```

Comment:
the scope for this is a bit "nitpicking" I find with only two difference cases :) upper this blocks perhaps ?

Reply:
Sorry, I don't follow. Do you mean I should move the `ssize_t n;` declaration out of the preprocessor condition (i.e. directly below the `size_t amount_to_read`)?

Reply:
The n variable yes

Reply:
Okay, moved it outside the preprocessor condition. I wasn't sure about that one myself.